packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -209,24 +209,36 @@ contract Bridge is EssentialContract, IBridge {
 
         if (block.timestamp >= invocationDelay + receivedAt) {
             delete proofReceipt[msgHash];
-            messageStatus[msgHash] = Status.RECALLED;
 
+            // Some cases recall might fail (e.g. USDC shuts down minting allowance)
+            bool success = true;
+            bytes memory data;
             // Execute the recall logic based on the contract's support for the
             // IRecallableSender interface
             if (_message.from.supportsInterface(type(IRecallableSender).interfaceId)) {
                 _storeContext(msgHash, address(this), _message.srcChainId);
 
-                // Perform recall
-                IRecallableSender(_message.from).onMessageRecalled{ value: _message.value }(
-                    _message, msgHash
-                );
+                bytes memory msgRecallPayLoad =
+                    abi.encodeCall(IRecallableSender.onMessageRecalled, (_message, msgHash));
+
+                // Perform recall but allow it to fail, where in such case fire a
+                // MessageRecallFailed event, where the event itself and the root cause could be
+                // case-by-case investigated while recall failed
+                (success, data) = _message.from.call{ value: _message.value }(msgRecallPayLoad);
 
                 // Must reset the context after the message call
                 _resetContext();
             } else {
                 _message.srcOwner.sendEtherAndVerify(_message.value);
             }
-            emit MessageRecalled(msgHash);
+
+            if (success) {
+                emit MessageRecalled(msgHash);
+                messageStatus[msgHash] = Status.RECALLED;
+            } else {
+                emit MessageRecallFailed(msgHash, data);
+                messageStatus[msgHash] = Status.RECALL_FAILED;
```

Comment:
what's the benefit of setting the status to RECALL_FAILED?

Reply:
> what's the benefit of setting the status to RECALL_FAILED?

So the benefit is: we will be able to reward people somehow later-on (source or dest) and can be trustless. (On-chain proveable) agains a specific messageStatus.

> Without this PR, if a message failed in recallMessage, people can try it again, now they cannot as the check here will fail:

Yes, but:
1. according to the issue description it can (and indeed) forever stuck. So people can do 100x it would always revert.
2. low chance that we every see a `recallMessage` let alone a failing recall message.

Reply:
my counter argument:
- rewarding based on failed tx history is a very centralized decision, so using events is good enough.
- recallMessage shall rarely fail, in case of failure, for example, a vault has a bug, I think people will prefer to have a chance to try it again after the bug is fixed, rather than making it fail permanetly.

Reply:
Okay, i closing it then for now.